qutebrowser/browser/downloads.py

```diff
@@ -600,7 +603,8 @@ def _set_filename(self, filename, *, force_overwrite=False):
                                                   os.path.expanduser('~'))
 
         self.basename = os.path.basename(self._filename)
-        last_used_directory = os.path.dirname(self._filename)
+        if remember_directory:
+            last_used_directory = os.path.dirname(self._filename)
```

Comment:
Hmm, looks like the test still passes even when reverting this, which shouldn't happen? Not sure why, though I haven't taken a deeper look.